docs/source/api/user-guide/combinators/compositors.rst

```diff
@@ -194,11 +194,166 @@ sum and count:
+sum and count:
+
+.. tab-set::
+
+    .. tab-item:: Python
+        :sync: python
+
+        .. code:: python
+
+            >>> print("dp sum:", qbl_fully_adaptive_comp(meas_sum))
+            dp sum: ...
+            >>> print("dp count:", qbl_fully_adaptive_comp(meas_count))
+            dp count: ...
+
+Now that we have submitted two queries, we can see that the privacy loss has increased commensurately:
+
+.. tab-set::
+
+    .. tab-item:: Python
+        :sync: python
+
+        .. code:: python
+
+            >>> qbl_fully_adaptive_comp.privacy_loss()
+            3.0
+
+Privacy Filter
+--------------
+You can convert any odometer into a measurement by setting an upper bound on the privacy loss.
+The following example converts the fully adaptive composition odometer into a privacy filter
+that rejects any query that would cause the privacy loss to exceed 2.0:
+
+.. tab-set::
+
+    .. tab-item:: Python
+        :sync: python
+
+        .. code:: python
+
+            >>> meas_fully_adaptive_comp = dp.c.make_privacy_filter(
+            ...     odom_fully_adaptive_comp,
+            ...     d_out=2.0,
+            ... )
+
+Privacy filters are measurements, meaning that they can be passed into :func:`make_composition <opendp.combinators.make_composition>`, 
+adaptive composition queryables, or into other combinators.
+However, they have the added benefit of not needing to specify privacy-loss parameters ahead-of-time.
+When the privacy filter (``meas_fully_adaptive_comp``) is invoked, 
+it still returns an *odometer queryable*, but this time the queryable will limit the overall privacy loss.
+
+.. tab-set::
+
+    .. tab-item:: Python
+        :sync: python
+
+        .. code:: python
+
+            >>> int_dataset = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+            >>> qbl_fully_adaptive_comp = meas_fully_adaptive_comp(int_dataset)
+            
+Similarly as before, we now interactively submit queries to estimate the
+sum and count:
+
+.. tab-set::
+
+    .. tab-item:: Python
+        :sync: python
+
+        .. code:: python
+
+            >>> print("dp count:", qbl_fully_adaptive_comp(meas_count))
+            dp count: ...
+            >>> print("dp count:", qbl_fully_adaptive_comp(meas_count))
+            dp count: ...
+
+Now that we have submitted two queries, we can see that the privacy loss has increased commensurately:
+
+.. tab-set::
+
+    .. tab-item:: Python
+        :sync: python
+
+        .. code:: python
+
+            >>> qbl_fully_adaptive_comp.privacy_loss()
+            2.0
+
+Since the privacy loss is capped at 2.0, any more queries will be rejected:
+
+.. tab-set::
+
+    .. tab-item:: Python
+        :sync: python
+
+        .. code:: python
+
+            >>> print("dp count:", qbl_fully_adaptive_comp(meas_count))
+            Traceback (most recent call last):
+            ...
+            opendp.mod.OpenDPException: 
+              FailedFunction("insufficient privacy budget: 3.0 > 2.0")
```

Comment:
In the error message, would it be worth separating the previous value (2.0) from the additional spend (1.0)? In this case you were already at the limit, but in other cases there might still be some budget left that could still be used if the last query were adjusted.